spacy/training/loop.py

```diff
@@ -217,17 +214,17 @@ def train_while_improving(
                 subbatch,
                 drop=dropout,
                 losses=losses,
-                sgd=False,  # type: ignore[arg-type]
+                sgd=None,  # type: ignore[arg-type]
                 exclude=exclude,
                 annotates=annotating_components,
             )
-            if use_rehearse:
-                nlp.rehearse(
-                    subbatch,
-                    losses=losses,
-                    sgd=False,  # type: ignore[arg-type]
-                    exclude=exclude,
-                )
+            nlp.rehearse(
+                subbatch,
+                losses=losses,
+                sgd=None,  # type: ignore[arg-type]
```

Comment:
```suggestion
                sgd=None,
```
I think we can remove the `mypy` hint now?

Reply:
I've seen this in a couple of other places too. My best guess: a compatibility hack to support both Python 2 and 3 (from the time we were supporting both).